gum/backend-darwin/gumprocess-darwin.c

```diff
@@ -532,22 +610,44 @@ gum_module_enumerate_ranges (const gchar * module_name,
     if (lc->cmd == GUM_LC_SEGMENT)
     {
       gum_segment_command_t * segcmd = (gum_segment_command_t *) lc;
+      gboolean is_page_zero;
       GumPageProtection cur_prot;
 
+      is_page_zero = segcmd->vmaddr == 0 &&
+          segcmd->filesize == 0 &&
+          segcmd->vmsize != 0 &&
+          (segcmd->initprot & VM_PROT_ALL) == VM_PROT_NONE &&
+          (segcmd->maxprot & VM_PROT_ALL) == VM_PROT_NONE;
+
+      if (is_page_zero) {
+          p += lc->cmdsize;
+          continue;
+      }
+
       cur_prot = gum_page_protection_from_mach (segcmd->initprot);
 
       if ((cur_prot & prot) == prot)
       {
         GumMemoryRange range;
         GumRangeDetails details;
+        GumFileMapping file;
+        struct proc_regionwithpathinfo reginfo;
```

Comment:
Might be worth renaming `reginfo` to `region_info` (or maybe just `region`), as we use `reg` as abbreviation for `register` in some other places.